# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/debugger/Script.h
# Commit: e802face8e04
# Full Hash: e802face8e042d03933c1495d36fd02a9def7dd9
# Author: Brian Hackett <bhackett1024@gmail.com>
# Date: 2019-07-25 21:51:57
# Regressor Bug: 1554524
# File Overlap Count: 1
# Description:
#   Bug 1554524 Part 4 - Add Debugger interface for instrumenting scripts, r=jimb,tcampbell.
# ==============================================================================

diff -r fc204daba6f3 -r e802face8e04 js/src/debugger/Script.h
--- a/js/src/debugger/Script.h	Mon Jul 08 09:39:52 2019 -1000
+++ b/js/src/debugger/Script.h	Mon Jul 08 09:41:21 2019 -1000
@@ -33,6 +33,10 @@
 
   enum {
     OWNER_SLOT,
+
+    // Holds any instrumentation ID that has been assigned to the script.
+    INSTRUMENTATION_ID_SLOT,
+
     RESERVED_SLOTS,
   };
 
@@ -84,6 +88,7 @@
   static bool clearAllBreakpoints(JSContext* cx, unsigned argc, Value* vp);
   static bool isInCatchScope(JSContext* cx, unsigned argc, Value* vp);
   static bool getOffsetsCoverage(JSContext* cx, unsigned argc, Value* vp);
+  static bool setInstrumentationId(JSContext* cx, unsigned argc, Value* vp);
   static bool construct(JSContext* cx, unsigned argc, Value* vp);
 
   template <typename T>
@@ -93,6 +98,10 @@
                                                Value* vp, const char* name,
                                                bool successor);
 
+  Value getInstrumentationId() const {
+    return getSlot(INSTRUMENTATION_ID_SLOT);
+  }
+
  private:
   static const ClassOps classOps_;
 